services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -0,0 +1,942 @@
+        def newProtectedPath
+        Wrappers.wait(WAIT_OFFSET) {
+            newProtectedPath = pathHelper.convert(northbound.getFlowPath(flow.id).protectedPath)
+            newProtectedPath != currentProtectedPath
+        }
+
+        and: "Bandwidth is reserved for new protected path on involved isls"
+        def newProtectedIsls = pathHelper.getInvolvedIsls(newProtectedPath)
+        def allLinks = northbound.getAllLinks()
+        def newProtectedIslsInfo = newProtectedIsls.collect { islUtils.getIslInfo(allLinks, it).get() }
+
+        allIsls.each { isl ->
+            newProtectedIslsInfo.each { protectedIsl ->
+                if (isl.id == protectedIsl.id) {
+                    assert isl.availableBandwidth - protectedIsl.availableBandwidth == flow.maximumBandwidth
+                }
+            }
+        }
+
+        and: "Reservation is deleted on the broken isl"
+        def originInfoBrokenIsl = allIsls.find {
+            it.destination.switchId == islToBreakProtectedPath.dstSwitch.dpId &&
+                    it.destination.portNo == islToBreakProtectedPath.dstPort
+        }
+        def currentInfoBrokenIsl = islUtils.getIslInfo(allLinks, islToBreakProtectedPath).get()
+
+        originInfoBrokenIsl.availableBandwidth == currentInfoBrokenIsl.availableBandwidth
+
+        when: "Restore port status"
+        northbound.portUp(islToBreakProtectedPath.dstSwitch.dpId, islToBreakProtectedPath.dstPort)
+        Wrappers.wait(WAIT_OFFSET + discoveryInterval) {
+            assert islUtils.getIslInfo(islToBreakProtectedPath).get().state == IslChangeType.DISCOVERED
+        }
+
+        then: "Path is not recalculated again"
+        pathHelper.convert(northbound.getFlowPath(flow.id).protectedPath) == newProtectedPath
+
+        and: "Cleanup: revert system to original state"
+        flowHelper.deleteFlow(flow.id)
+        database.resetCosts()
+    }
+
+    @Unroll
+    def "Unable to create protected path on #flowDescription flow on the same path"() {
+        given: "Two active neighboring switches with two not overlapping paths at least"
+        def switches = topology.getActiveSwitches()
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            database.getPaths(src.dpId, dst.dpId)*.path.collect {
+                pathHelper.getInvolvedIsls(it)
+            }.unique { a, b -> a.intersect(b) ? 0 : 1 }.size() >= 2
+        } ?: assumeTrue("No suiting switches found", false)
+
+        and: "A flow without protected path"
+        def flow1 = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow1.allocateProtectedPath = false
+        flowHelper.addFlow(flow1)
+
+        and: "All alternative paths are unavailable (bring ports down on the source switch)"
+        List<PathNode> broughtDownPorts = []
+        database.getPaths(srcSwitch.dpId, dstSwitch.dpId)*.path.findAll { it != flow1 }.unique {
```

Comment:
`it != flow1`
You try to compare path and flow objects.

Reply:
fixed